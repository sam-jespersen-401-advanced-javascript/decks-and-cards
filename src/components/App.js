import React from 'react';
import { characters, gifs } from '../content.json';

export default function App() {
  return (
    <>
      <section>
        <header>
          <h2>Characters</h2>
        </header>
        {characters.map(({ _id, name, image }) => (
          <div key={_id}>
            <header><h3>{name}</h3></header>
            <figure>
              <img src={image} alt={name} />
            </figure>
            <p>{name} has {_id}</p>
          </div>
        ))}
      </section>
      <section>
        {gifs.map(({ _id, image }) => (
          <div key={_id}>
            <header><h3>{_id}</h3></header>
            <figure>
              <img src={image} alt={_id} />
            </figure>
          </div>
        ))}
      </section>
    </>
  );
}
