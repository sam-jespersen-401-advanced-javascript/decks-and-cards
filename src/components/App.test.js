import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

describe('App component', () => {
  it('renders App', () => {
    const wrapper = shallow(<App count={12} />);
    expect(wrapper).toMatchSnapshot();
  });
});
