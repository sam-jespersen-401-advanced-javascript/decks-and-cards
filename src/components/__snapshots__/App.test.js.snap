// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<Fragment>
  <section>
    <Header />
    <Characters
      count={12}
    />
    <Gifs
      count={12}
    />
  </section>
</Fragment>
`;
